contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -2987,3 +2994,22 @@ pltsql_planner_hook(Query *parse, const char *query_string, int cursorOptions, P
 
 	return plan;
 }
+
+static Node* 
+transform_like_in_add_constraint (Node* node)
+{
+	bool create_tbl_stmt_is_invalid = false;
+
+	if (current_query_is_create_tbl_check_constraint)
+		create_tbl_stmt_is_invalid = expr_contains_ilike_and_ci_collation_wrapper(node, true);
+
+	if (!babelfish_dump_restore && current_query_is_create_tbl_check_constraint && create_tbl_stmt_is_invalid)
+	{
+		current_query_is_create_tbl_check_constraint = false;
+		ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					errmsg("nondeterministic collations are not supported for ILIKE")));
+	}
```

Comment:
Do we have a test case for this?

Reply:
We don't have a test case for this because this check is for psql syntax. ILIKE is not supported in tsql syntax.

Reply:
Added tests in .mix files